pkg/clusteragent/custommetrics/provider.go

```diff
@@ -45,13 +45,61 @@ type datadogProvider struct {
 // NewDatadogProvider creates a Custom Metrics and External Metrics Provider.
 func NewDatadogProvider(client dynamic.Interface, mapper apimeta.RESTMapper, store Store) provider.MetricsProvider {
 	maxAge := config.Datadog.GetInt64("external_metrics_provider.local_copy_refresh_rate")
-	return &datadogProvider{
+	d := &datadogProvider{
 		client: client,
 		mapper: mapper,
-		values: make(map[provider.CustomMetricInfo]int64),
 		store:  store,
 		maxAge: maxAge,
 	}
+	go d.externalMetricsSetter()
+	return d
+}
+
+func (p *datadogProvider) externalMetricsSetter() {
```

Comment:
How do you exist from this method in a normal case? 
Maybe we should pass in argument a `context.Context` to received a `Done()` event if needed for a graceful stop.

Reply:
I'm actually still debating on how to best do this. 
context is a good idea, I can get it from the parent method